File: 11기 코딩테스트_230308/연습3/CountDistinctLetter/CountDistinctLetter_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.CountDistinctLetter.CountDistinctLetter import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'google'

        # when
        result = solution(s)

        # then
        self.assertEqual(4, result)

    def test_2(self):
        # given
        s = 'navigate'

        # when
        result = solution(s)

        # then
        self.assertEqual(7, result)

    def test_3(self):
        # given
        s = 'window'

        # when
        result = solution(s)

        # then
        self.assertEqual(5, result)

    def test_4(self):
        # given
        s = ''

        # when
        result = solution(s)

        # then
        self.assertEqual(0, result)

    def test_5(self):
        # given
        s = 'z'

        # when
        result = solution(s)

        # then
        self.assertEqual(1, result)

    def test_6(self):
        # given
        s = 'problem'

        # when
        result = solution(s)

        # then
        self.assertEqual(7, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습3/CountDistinctLetter/CountDistinctLetter_test.js
import assert from 'assert'
import solution from './CountDistinctLetter.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'google'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(4, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'navigate'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(7, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'window'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(5, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = ''

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(0, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'z'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(1, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'problem'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(7, result)
    done();
  });
})
